### PR TITLE
Update workspace workflow tests to not override existing workflows in the policy.

### LINKED ISSUE #95: Workflowtest aktualisieren
Beim Ausführen des Tests https://github.com/4teamwork/bobtemplates.4teamwork/blob/master/bobtemplates/workspace/%2Bpackage.fullname%2B/%2Bpackage.part_1%2B/%2Bpackage.part_2%2B/tests/test_workflows.py.bob#L7 wird das spez.txt von der Policy mit dem spez.txt von `ftw.workspace` ersetzt.

`ftw.workspace` definiert jeweils einen "empfohlenen Workflow". Dieser Workflow muss einmalig in die Policy kopiert werden weil die Plicy und nicht das Modul den Workflow steuert. Wird nun ein Workflow aufgrund einer Änderung im Modul `ftw.workspace` geändert sollten diese Änderungen auch in der Policy aktualisiert werden. Dies zu erkennen ist jedoch schwierig und viele Maintainer würden die Workflows ihrer Policies nicht updaten.

Damit diese Änderungen erkannt werden, wurde der Workflow-Test so erweitert, dass er jeweils automatisch das aktuelle spez.txt vom `ftw.workspace` in die Policy kopiert.

Dies hat den Vorteil, dass sich der Maintainer nicht um die Aktualität des spez.txt in seiner Policy kümmern muss.

Leider bringt diese Implementation auch Probleme bzw. Verwirrungen mit sich. Will man sein Policy-Spez.txt nun ändern und führt anschliessend die Tests aus, werden diese Änderungen wieder überschrieben.

Aus diesem Grund müssen wir den Test überarbeiten und "sicherer" machen:
- [x] Test soll nichts mehr automatisch überschreiben
- [x] Test soll nur vergleichen, ob das spez.txt noch aktuell ist
- [x] Ist das spez.txt nicht mehr aktuell, soll eine klare Meldung ausgegeben werden mit folgenden Infos:
  - [x] Genaues diff der Files
  - [x] Erklärung, warum der Test genau failed und warum dies gleich sein sollte, bzw. warum dieser Test überhaupt ausgeführt wird
  - [x] Pfade zu beiden spez.txt damit diese schnell kopiert werden können (evt. direkt der command zum kopieren ausgeben gem. git)
  - [x] Erklärung wie der Test abgeändert werden muss wenn man den Workflow bewusst geändert hat

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR will update the workspace template workflow test.

For more details see issue #95 

closes #95

### CODE REVIEW

File: bobtemplates/workspace/+package.fullname+/+package.part_1+/+package.part_2+/tests/test_workflows.py.bob [hunk 1]
```diff
@@ -2,34 +2,83 @@ from ftw.lawgiver.tests.base import WorkflowTest
...
+import os
 
 
-class CopyFromWorkspaceWorkflowTest(WorkflowTest):
+class CompareWorkspaceWorkflowTest(WorkflowTest):
     layer = {{{package.part_1_upper}}}_FUNCTIONAL
 
+    specification_name = 'specification.de.txt'
+
     def setUp(self):
```
Comment (jone) [inline bobtemplates/workspace/+package.fullname+/+package.part_1+/+package.part_2+/tests/test_workflows.py.bob]: Before, the setup did copy the spec, which was a precondition of the other tests, thus it was in setup.
Now its just a regular test / assertion and does not do things, right?

I'd therefore move that to a `test_` method and remove the `setUp` altogether.

File: bobtemplates/workspace/+package.fullname+/+package.part_1+/+package.part_2+/tests/test_workflows.py.bob [hunk 2]
```diff
@@ -2,34 +2,83 @@ from ftw.lawgiver.tests.base import WorkflowTest
...
+        if self._is_base_test():
+            return
+
+        spec_source_path = Path(ftw.workspace.__file__).joinpath(
+            '..', 'profiles', 'default',
+            'workflows', Path(self.workflow_path).basename(),
+            self.specification_name).abspath()
+
+        self.assertTrue(
+            os.path.isfile(spec_source_path),
```
Comment (jone) [inline bobtemplates/workspace/+package.fullname+/+package.part_1+/+package.part_2+/tests/test_workflows.py.bob]: The Path-object has methods for such things:

```
spec_source_path.isfile()
```

See http://pythonhosted.org/path.py/api.html#path.Path.isfile 😉

File: bobtemplates/workspace/+package.fullname+/+package.part_1+/+package.part_2+/tests/test_workflows.py.bob [hunk 3]
```diff
@@ -2,34 +2,83 @@ from ftw.lawgiver.tests.base import WorkflowTest
...
+            "The source file at '{}' does no longer exists.".format(
+                spec_source_path))
+
+        spec_source = open(spec_source_path).read()
```
Comment (jone) [inline bobtemplates/workspace/+package.fullname+/+package.part_1+/+package.part_2+/tests/test_workflows.py.bob]:
``` diff
-        spec_source = open(spec_source_path).read()
+        spec_source = spec_source_path.bytes()
```

😉

**Comment by jone** [comment]
Awesome, I like such tests 😊 

Just a thought: Should we move that to `ftw.workspace` and just inherit here?
This would make it easier to apply changes since we are not copying the code..

**Comment by elioschmutz** [comment]
👍  thats a good idea! I'll move the test.

**Comment by elioschmutz** [comment]
@jone 👍  everything works fine and tests are passing.

Moved CompareWorkspaceWorkflowTest to https://github.com/4teamwork/ftw.workspace/commit/ca47cd09533904b63f8817c02a1e5b7b6c1d912a

**Pull request merged**